src/main/java/hlf/java/rest/client/service/impl/ChannelServiceImpl.java

```
@@ -0,0 +1,488 @@
+      }
+
+      // add certificates in the request to keystore
+      addToKeyStore(channelOperationRequest);
+
+      for (Peer peer : peers) {
+        Set<String> channels = hfClient.queryChannels(peer);
+        if (!channels.contains(channelName)) {
+          channel.addOrderer(orderer);
+          channel.joinPeer(peer);
+        }
+      }
+
+      return new ClientResponseModel(
+          ChannelOperationStatus.CODE_SUCCESS, ChannelOperationStatus.SUCCESS_JOINED + channelName);
+    } catch (Exception e) {
+      log.error("Error in joining channel ", e);
+      String errorMsg = e.getMessage();
+      errorMsg = errorMsg.length() > 50 ? errorMsg.substring(0, 50) + "..." : errorMsg;
+      if (channel != null) {
+        channel.shutdown(false);
+      }
+      throw new ChannelOperationException(ErrorCode.FABRIC_CHANNEL_JOIN_FAILED, errorMsg);
+    }
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public Set<String> listChannels(String peerName, String peerGrpcUrl) {
+    try {
+      Peer peer = hfClient.newPeer(peerName, peerGrpcUrl);
+      return hfClient.queryChannels(peer);
+    } catch (InvalidArgumentException e) {
+      log.error("Error in retrieving peer: ", e);
+    } catch (ProposalException e) {
+      log.error("Error in querying channels: ", e);
+    }
+    return new HashSet<>();
+  }
+
+  /**
+   * create new channel configuration
+   *
+   * @param channelOperationRequest
+   * @return channel configuration
+   * @throws Exception
+   */
+  private ChannelConfiguration newChannelConfig(ChannelOperationRequest channelOperationRequest)
+      throws Exception {
+    ChannelConfiguration newChannelConfig = new ChannelConfiguration();
+    try {
+      Common.Envelope newChannelEnvelop = getChannelCreationTransaction(channelOperationRequest);
+      ;
+      newChannelConfig.setChannelConfiguration(newChannelEnvelop.toByteArray());
+    } catch (InvalidArgumentException e) {
+      e.printStackTrace();
```

Should throw an exception instead.